sympy/polys/rings.py

```diff
@@ -2987,34 +3148,26 @@ def _compose(self, replacements, initial_poly):
 
         return poly
 
-    def _div(self, fv):
-        # Implement the same algorithm from [CLO] p64. in python-flint
+    # XXX: implement the same algorith for div from CLO
+    # for python-flint
+    def _div(self, fv: PolyElement[Er]) -> tuple[PolyElement[Er], PolyElement[Er]]:
+        [q], r = self._div_list([fv])
+        return q, r
+
+    def _div_list(self, fv: list[PolyElement[Er]]) -> tuple[list[PolyElement[Er]], PolyElement[Er]]:
         ring = self.ring
-        ret_single = False
-        if isinstance(fv, PolyElement):
-            ret_single = True
-            fv = [fv]
-        if not all(fv):
-            raise ZeroDivisionError("polynomial division")
-        if not self:
-            if ret_single:
-                return ring.zero, ring.zero
-            else:
-                return [], ring.zero
-        for f in fv:
-            if f.ring != ring:
-                raise ValueError("self and f must have the same ring")
         s = len(fv)
         qv = [ring.zero for i in range(s)]
         p = self.copy()
         r = ring.zero
         term_div = self._term_div()
-        expvs = [fx.leading_expv() for fx in fv]
+        expvs = [cast('Mon', fx.leading_expv()) for fx in fv]
```

Comment:
Rather than using cast here it would be better to have a method that returns the expected type:
```python
def _leading_expv(self) -> Mon:
    if not self:
        raise KeyError
    else:
        return ...

def leading_expv(self) -> Mon | None:
    try:
        return self._leading_expv()
    except KeyError:
        return None
```
Then internal code can use `_leading_expv`.